WordPress/gutenberg
# Style engine: switch off optimize by default

Resolves https://github.com/WordPress/gutenberg/issues/52646

## What?
The style engine's default behavior of combining selectors where the CSS declarations are equal results in errors where order matters.

This PR doesn't address the core issue, that is, maintaining the order of general selectors where CSS stylesheet opt-in to optimization. 

I'm not convinced that it can given how general elements rules are. Optimization should therefore be "opt-in" only.

For example, many blocks, children of children of parents or whichever, can have the following selector applied to them: `.wp-element-1 a`. It's impossible for the style engine to know the intended hierarchy when combining such selectors with matching declarations. cc @aristath for thoughts.

If we want the optimizations to work universally, I think elements block supports should create rules specific to the block that needs them. How, I'm not sure. Maybe if we're applying a link color to a paragraph block we could do something like `p.wp-element-1 > a`. 🤷🏻 


## Why?
See: https://core.trac.wordpress.org/ticket/58811
Core patch: https://github.com/WordPress/wordpress-develop/pull/4902

The bug is most evident in elements block supports CSS (links), which applies CSS rules to block wrappers that affect all child elements.

If a child element and another parent element (not the child's) share CSS declarations, the CSS selectors will be combined and placed towards of the end of the stylesheet thereby overriding any previous rules.

So far it's only occurring for elements (link colors) but it potentially affects any selector that has general application, e.g., `.block-wrapper-class a`

The bug is only visible on the frontend (not the editor).

## How?
Flipping the `optimize` flag and updating tests

## Testing Instructions

Follow the test instructions in https://github.com/WordPress/gutenberg/issues/52646



<details>

<summary>Here is some test HTML:</summary>

```html

<div class="wp-block-group has-luminous-vivid-orange-color has-tertiary-background-color has-text-color has-background has-link-color" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)">
<div class="wp-block-group has-vivid-red-color has-text-color has-link-color">
<p>A paragraph <a href="https://wordpress.org" data-type="link" data-id="https://wordpress.org">inside two group blocks with a link</a></p>
</div>
</div>



<div class="wp-block-group has-vivid-purple-color has-tertiary-background-color has-text-color has-background has-link-color" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)">
<div class="wp-block-group has-luminous-vivid-orange-color has-text-color has-link-color">
<p>A paragraph <a href="https://wordpress.org" data-type="link" data-id="https://wordpress.org">inside two group blocks with a link</a></p>
</div>
</div>


```


</details>

Using the above code you should see the following on the frontend


| Before  | After |
| ------------- | ------------- |
| <img width="400" alt="Screenshot 2023-07-28 at 11 34 50 am" src="https://github.com/WordPress/gutenberg/assets/6458278/f6903f43-36e1-4a28-b783-3cda3346a6bc">  | <img width="400" alt="Screenshot 2023-07-28 at 11 34 55 am" src="https://github.com/WordPress/gutenberg/assets/6458278/be503e5e-8522-40b3-a5ca-bdaad34a6bf0">  |
